open-sphere-plugins/wfs/src/main/java/io/opensphere/wfs/state/activate/WFSStateActivator.java

```diff
@@ -100,60 +110,37 @@ public WFSStateGroup activateState(String id, StateType state) throws Interrupte
 
         List<WFSLayerState> states = ServerStateUtilities.getWfsLayers(serverToolbox.getLayerConfigurationManager(), state)
                 .stream().map(this::convertLayer).collect(Collectors.toList());
-        List<DataGroupInfo> stateGroups = createAndActivateGroups(id, states);
+        List<DataGroupInfo> stateGroups = getGroupsFromStates(states);
+
+        activateGroups(stateGroups);
 
         return new WFSStateGroup(id, stateGroups, states);
     }
 
     /**
-     * Converts the state layer to a WFSLayerState.
+     * Activates the data groups if any of them are inactive.
      *
-     * @param layerType the state layer
-     * @return the WFSLayerState
+     * @param groups the data groups to activate
+     * @throws InterruptedException if activation is interrupted
      */
-    protected WFSLayerState convertLayer(LayerType layerType)
+    protected void activateGroups(List<DataGroupInfo> groups) throws InterruptedException
     {
-        return SaveStateV4ToV3Translator.toLayerState(layerType);
+        if (groups.stream().anyMatch(e -> e.activationProperty().isInactiveOrDeactivation()))

```

Comment:
Is this correct that you want to check if ANY group is inactive and then activate all groups?  Or do you want to activate each inactive group?

Reply:
Changed to activate groups individually, removed stream to avoid checked exception weirdness.